File: prism-sdk/credentials/jvm/src/main/java/io/iohk/atala/prism/credentials/japi/KeyData.java
package io.iohk.atala.prism.credentials.japi;

import io.iohk.atala.prism.crypto.japi.ECPublicKey;

import java.util.Optional;

public class KeyData {
    // TODO: DO NOT USE ECPublicKey
    private final ECPublicKey publicKey;
    private final TimestampInfo addedOn;
    private final Optional<TimestampInfo> revokedOn;

    public KeyData(ECPublicKey publicKey,
                   TimestampInfo addedOn,
                   Optional<TimestampInfo> revokedOn) {
        this.publicKey = publicKey;
        this.addedOn = addedOn;
        this.revokedOn = revokedOn;
    }

    public ECPublicKey getPublicKey() {
        return publicKey;
    }

    public TimestampInfo getAddedOn() {
        return addedOn;
    }

    public Optional<TimestampInfo> getRevokedOn() {
        return revokedOn;
    }
}

File: prism-kotlin-sdk/credentials/src/commonMain/kotlin/io/iohk/atala/prism/kotlin/credentials/KeyData.kt
package io.iohk.atala.prism.kotlin.credentials

import io.iohk.atala.prism.kotlin.crypto.keys.ECPublicKey
import kotlin.js.JsExport

@JsExport
data class KeyData(
    val publicKey: ECPublicKey,
    val addedOn: TimestampInfo,
    val revokedOn: TimestampInfo?
)
